Initium-Odp/src/com/universeprojects/miniup/server/commands/CommandSetLabel.java

```diff
@@ -0,0 +1,91 @@
+package com.universeprojects.miniup.server.commands;
+
+import java.util.Map;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.universeprojects.cacheddatastore.CachedDatastoreService;
+import com.universeprojects.cacheddatastore.CachedEntity;
+import com.google.appengine.api.datastore.Key;
+import com.universeprojects.miniup.server.ODPDBAccess;
+import com.universeprojects.miniup.server.WebUtils;
+import com.universeprojects.miniup.server.commands.framework.Command;
+import com.universeprojects.miniup.server.commands.framework.UserErrorMessage;
+
+
+/**
+ * Allows the current leader of a party to set a different party member as leader
+ * 
+ * Usage notes:
+ * Checks if itemId is a valid storage option, and in inventory/at the current location
+ * Also sanitises label for any possible injection ideas. 
+ * 
+ * Parameters:
+ * 		itemId - itemID of the chest to be relabelled 
+ * 		label - the new label
+ * 
+ * @author NJ
+ *
+ */
+public class CommandSetLabel extends Command {
+
+	public CommandSetLabel(HttpServletRequest request, HttpServletResponse response) 
+	{
+		super(request, response);
+	}
+
+	@Override
+	public void run(Map<String, String> parameters) throws UserErrorMessage 
+	{
+		// Get the DB access object. We use this for most DB communication. It is a place for reusable methods that access the DB.
+		ODPDBAccess db = getDB();
+
+		// Check if we're logged in or not. We shouldn't be able to see this since commands cannot be used unless the user is logged in.
+		if (db.isLoggedIn(request)==false)
```

Comment:
Ok don't worry about it then. I'll see what I can do when I have a chance.
Probably just need to update the javascript a bit.

On Wed, Apr 13, 2016 at 8:41 AM, NikeJoshua <notifications@github.com>
wrote:

> In
> Initium-Odp/src/com/universeprojects/miniup/server/commands/CommandSetLabel.java
> <https://github.com/Emperorlou/initium-odp/pull/5#discussion_r59540080>:
>
> > + */
> > +public class CommandSetLabel extends Command {
> > +
> > +	public CommandSetLabel(HttpServletRequest request, HttpServletResponse response)
> > +	{
> > +		super(request, response);
> > +	}
> > +
> > +	@Override
> > +	public void run(Map<String, String> parameters) throws UserErrorMessage
> > +	{
> > +		// Get the DB access object. We use this for most DB communication. It is a place for reusable methods that access the DB.
> > +		ODPDBAccess db = getDB();
> > +
> > +		// Check if we're logged in or not. We shouldn't be able to see this since commands cannot be used unless the user is logged in.
> > +		if (db.isLoggedIn(request)==false)
>
> Just tested with a manually created command URL when logged out, and
> indeed gets thrown prior to this check, so will remove these for next
> update. The output is a regular errorMessage json, but since people aren't
> logged in, there's no clean handler for this and it just gets displayed as
> such:
> {"errorMessage":"You are not currently logged in. \u003ca
> href\u003d\u0027login.jsp\u0027\u003eClick here\u003c\/a\u003e to log in
> before doing anything else."}
> Maybe better to just write direct html output here?
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/Emperorlou/initium-odp/pull/5/files/2efff2122e588bd8ab06866193b3061969636da8#r59540080>
>